nextmove/src/main/java/no/difi/meldingsutveksling/nextmove/DpoConversationResource.java

```diff
@@ -1,27 +1,49 @@
 package no.difi.meldingsutveksling.nextmove;
 
-import com.google.common.collect.Maps;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 import no.difi.meldingsutveksling.ServiceIdentifier;
 
 import javax.persistence.DiscriminatorValue;
 import javax.persistence.Entity;
 import java.time.LocalDateTime;
+import java.util.Map;
 
 @Entity
 @DiscriminatorValue("DPO")
+@NoArgsConstructor
 @Data
 public class DpoConversationResource extends ConversationResource {
 
-    private String jpId;
-
-    DpoConversationResource() {}
-
     private DpoConversationResource(String conversationId, String senderId, String receiverId) {
-        super(conversationId, senderId, receiverId, ServiceIdentifier.DPO, LocalDateTime.now(), Maps.newHashMap(), Maps.newHashMap());
+        super(conversationId, senderId, receiverId, ServiceIdentifier.DPO);
+    }
+
+    private DpoConversationResource(String conversationId,
+                                    String senderId,
+                                    String senderName,
+                                    String receiverId,
+                                    String receiverName,
+                                    LocalDateTime lastUpdate,
+                                    Map fileRefs,
+                                    Map customProperties) {
+        super(conversationId, senderId, senderName, receiverId, receiverName, ServiceIdentifier.DPO, lastUpdate, fileRefs, customProperties);
     }
 
     public static DpoConversationResource of(String conversationId, String senderId, String receiverId) {
         return new DpoConversationResource(conversationId, senderId, receiverId);
     }
+
+    public static DpoConversationResource of(ConversationResource cr) {
+        DpoConversationResource dpo = new DpoConversationResource(cr.getConversationId(),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "dpo". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1488)